docs/preview/features/endpoint-validation.md

```diff
@@ -3,53 +3,101 @@ title: "Endpoint validation"
 layout: default
 ---
 
-## Endpoint validation
+# Endpoint validation
 
-![](https://img.shields.io/badge/Available%20starting-v1.0-green)
+We provide support for endpoint validation, when implementing your own custom web hook. This validation allows to secure your web hook with a secret key (taken from the query string or an HTTP header).  
+This is needed, because Azure Event Grid is send a validation request to a newly configured web hook, in order to prevent people leveraging Azure Event Grid to bring down a 3rd party API. 
 
-We provide support for endpoint validation, when implementing your own custom web hook.  This validation allows to secure your web hook with a secret key (taken from the query string or an HTTP header).  This is needed, because Azure Event Grid is send a validation request to a newly configured web hook, in order to prevent people leveraging Azure Event Grid to bring down a 3rd party API.  The implementation we provide, is echoing back the validation key on your operation, in order to have the validation by Event Grid out of the box.
+## Installation
 
-Import the following namespace into your project:
-```csharp
-using Arcus.EventGrid.Security;
-using Arcus.EventGrid.Security.Attributes;
+The features described here require the following package:
+
+```shell
+PM> Install-Package Arcus.EventGrid.WebApi.Security 
 ```
 
-Next, tag the Operation that will be handling the Event Grid events with the `EventGridSubscriptionValidator` and the  `EventGridAuthorization` attributes, that will secure the endpoint and handle the handshake.
+## Usage
+
+The implementation we provide, is echoing back the validation key on your operation, in order to have the validation by Event Grid out of the box.
+
+### Enforce authorization globally
+
+We created the `EventGridAuthorizationFilter` MVC filter that will secure the endpoint and handle the handshake.
+
 ```csharp
-[EventGridSubscriptionValidator]
-[EventGridAuthorization("x-api-key", "event-grid")]
-public IHttpActionResult HandleEvents(HttpRequestMessage message)
+using Arcus.EventGrid.WebApi.Security;
+using Microsoft.Extensions.DependencyInjection;
+
+public class Startup
 {
-    return Ok();
+    public void ConfigureService(IServiceCollection services)
+    {
+        // Default set of authorization options.
+        var options = new EventGridAuthorizationOptions();
+
+        // Looks for the 'x-api-key' header in the HTTP request and tries to match it with the secret retrieved in the secret store with the name 'MySecret'.
+        services.AddMvc(options => options.Filters.Add(new EventGridAuthorizationFilter(HttpRequestProperty.Header, "x-api-key", "MySecret", options)));
```

Comment:
This makes me think if we shouldn't add an additional extension on the `IFilterCollection` to let the user configure these options inside a lambda.

Reply:
Added them, resolving.